services/web-server/src/utils/login.js

```diff
@@ -1,5 +1,17 @@
+const assert = require('assert');
+const _ = require('lodash');
+
 // A function that returns a middleware that renders callback.ejs
 module.exports = (publicUrl) => async (request, response) => {
+  assert(request.user, 'Must have a user');
+  assert(request.user.profile);
+  assert(request.user.identityProviderId);
+  assert(request.user.providerExpires);
+
+  if (!_.isString(request.user.profile)) {
+    request.user.profile = new Buffer.from(JSON.stringify(request.user.profile)).toString('base64');
```

Comment:
I'm not familiar w/ JS. Can this be assigned to ``request.user.encodedProfile`` instead of overwriting ``user.profile``? The same name makes it look like `callback.ejs` has the same bug as before.

Reply:
Thank you to all my participating friends who helped me.